File: Behavioral/Visitor/Java/ExportVisitor.java
// ExportVisitor.java

public class ExportVisitor implements ShapeVisitor {
  @Override
  public void visit(Circle circle) {
    System.out.println("Exporting a Circle to file.");
  }

  @Override
  public void visit(Square square) {
    System.out.println("Exporting a Square to file.");
  }
}

File: Behavioral/Visitor/JavaScript/exportVisitor.js
// exportVisitor.js

const ShapeVisitor = require("./shapeVisitor");

class ExportVisitor extends ShapeVisitor {
  visitCircle(circle) {
    console.log("Exporting a Circle to file.");
  }

  visitSquare(square) {
    console.log("Exporting a Square to file.");
  }
}

module.exports = ExportVisitor;
